src/Core/src/Handlers/View/ViewHandler.cs

```diff
@@ -15,35 +15,126 @@
 
 namespace Microsoft.Maui.Handlers
 {
-	public partial class ViewHandler
+	public abstract partial class ViewHandler : IViewHandler
 	{
 		public static PropertyMapper<IView> ViewMapper = new PropertyMapper<IView>
 		{
 			[nameof(IView.BackgroundColor)] = MapBackgroundColor,
 			[nameof(IView.Frame)] = MapFrame,
 			[nameof(IView.IsEnabled)] = MapIsEnabled,
-			[nameof(IView.AutomationId)] = MapAutomationId
+			[nameof(IView.AutomationId)] = MapAutomationId,
+			[nameof(IView.Semantics)] = MapSemantics
 		};
 
+
+		bool _hasContainer;
+		private NativeView? _view;
+		private IView? _virtualView;
+
+		public bool HasContainer
+		{
+			get => _hasContainer;
+			set
+			{
+				if (_hasContainer == value)
+					return;
+
+				_hasContainer = value;
+
+				if (value)
+					SetupContainer();
+				else
+					RemoveContainer();
+			}
+		}
+
+		protected abstract void SetupContainer();
+
+		protected abstract void RemoveContainer();
+
+		public IMauiContext? MauiContext { get; private set; }
+
+		public NativeView? View
```

Comment:
Maybe this should be named `NativeView` still. Just to avoid any confusion whatsoever.